torch/_refs/linalg/__init__.py

```diff
@@ -70,11 +70,13 @@ def vector_norm(
     # Checks
     check_fp_or_complex(x.dtype, "linalg.vector_norm")
 
+    if dim is None:
```

Comment:
would it be better to avoid `flatten` and set `dim=range(ndim)`? Flatten complicates the indexing if input was permuted.

Reply:
Yes, flatten was a bad idea here. I got rid of this in a subsequent commit already.

Reply:
Sorry, previous comment wasn't true when I said it, but it is now. (Very strange, I was positive I committed that change already)